### PR TITLE
cleanup(angular): refactor library generator to use nx devkit

### PR DESCRIPTION
**Pull request by Coly010**
Refactor Library Generator to use Nx Devkit

## Current Behavior

Library schematic uses angular devkit

## Expected Behavior

Library generator should use nx devkit

## Related Issue(s)

Fixes #

### CODE REVIEW

File: packages/angular/src/generators/move/lib/update-module-name.spec.ts [hunk 1]
```diff
@@ -114,118 +128,4 @@ describe('updateModuleName Rule', () => {
       );
     });
   });
-
-  describe('rename', () => {
```
Comment (leosvelperez) [inline packages/angular/src/generators/move/lib/update-module-name.spec.ts:118]: Why is this test suite removed? Is it covered anywhere else?

File: packages/angular/src/generators/move/move.spec.ts [hunk 2]
```diff
@@ -1,18 +1,25 @@
 import { Tree } from '@nrwl/devkit';
-import { wrapAngularDevkitSchematic } from '@nrwl/devkit/ngcli-adapter';
 import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
 import { angularMoveGenerator } from './move';
-
-const libSchematic = wrapAngularDevkitSchematic('@nrwl/angular', 'lib');
+import libraryGenerator from '../library/library';
+import { Linter } from '@nrwl/workspace';
```
Comment (leosvelperez) [inline packages/angular/src/generators/move/move.spec.ts:5]: This should be imported from `@nrwl/linter`.

File: packages/angular/src/generators/move/lib/update-module-name.spec.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
 import { Tree } from '@nrwl/devkit';
-import { wrapAngularDevkitSchematic } from '@nrwl/devkit/ngcli-adapter';
 import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
 import { moveGenerator } from '@nrwl/workspace';
 import { Schema } from '../schema';
 import { updateModuleName } from './update-module-name';
-
-const libSchematic = wrapAngularDevkitSchematic('@nrwl/angular', 'lib');
+import libraryGenerator from '../../library/library';
+import { Linter } from '@nrwl/workspace';
```
Comment (leosvelperez) [inline packages/angular/src/generators/move/lib/update-module-name.spec.ts:7]: This should be imported from `@nrwl/linter`.

File: packages/angular/src/generators/utils/testing.ts [hunk 4]
```diff
@@ -1,13 +1,15 @@
 import type { Tree } from '@nrwl/devkit';
 import { wrapAngularDevkitSchematic } from '@nrwl/devkit/ngcli-adapter';
 import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
+import { Linter } from '@nrwl/workspace';
```
Comment (leosvelperez) [inline packages/angular/src/generators/utils/testing.ts:4]: This should be imported from `@nrwl/linter`.

File: packages/workspace/src/generators/move/lib/move-project.ts [hunk 5]
```diff
@@ -15,11 +15,8 @@ export function moveProject(
 ) {
   const destination = getDestination(tree, schema, project);
   visitNotIgnoredFiles(tree, project.root, (file) => {
-    // This is a rename but Angular Devkit isn't capable of writing to a file after it's renamed so this is a workaround
-    const content = tree.read(file);
```
Comment (leosvelperez) [inline packages/workspace/src/generators/move/lib/move-project.ts:19]: I think it's safer to revert this change, the Angular DevKit doesn't support yet to rename and update a file, this workaround still applies.

**Review by leosvelperez — CHANGES_REQUESTED** [review]

File: packages/angular/src/generators/library/lib/update-project.ts [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+      compilerOptions: {
+        ...json.compilerOptions,
+        outDir: `${offsetFromRoot(options.projectRoot)}dist/out-tsc`,
+      },
+    };
+  });
+}
+
+function updateNxConfigWithProject(host: Tree, options: NormalizedSchema) {
+  return new Promise((resolve) => {
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/generators/library/lib/update-project.ts:177]: Ah it is

File: packages/workspace/src/generators/move/lib/move-project.ts (hunk 5 above)
Comment (Coly010) [reply packages/workspace/src/generators/move/lib/move-project.ts:19]: Ok. I had made the change as I had assumed that this would only ever be used by Nx Devkit but I guess someone can still run this as a schematic

File: packages/angular/src/generators/move/lib/update-module-name.spec.ts (hunk 1 above)
Comment (Coly010) [reply packages/angular/src/generators/move/lib/update-module-name.spec.ts:118]: I'm not actually sure. I'll double check

File: packages/angular/src/generators/library/library.ts [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    publishable: false,
+    simpleModuleName: false,
+    skipFormat: false,
+    unitTestRunner: UnitTestRunner.Jest,
+    ...schema,
+  };
+  // Do some validation checks
+  const options = normalizeOptions(host, mergedSchema);
+  if (!options.routing && options.lazy) {
+    throw new Error(`routing must be set`);
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/generators/library/library.ts:44]: Yep, I like that

File: packages/angular/src/generators/library/library.ts [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+  const runAngularLibrarySchematic = wrapAngularDevkitSchematic(
+    '@schematics/angular',
+    'library'
+  );
+  await runAngularLibrarySchematic(host, {
+    name: options.name,
+    prefix: options.prefix,
+    entryFile: 'index',
+    skipPackageJson: !(options.publishable || options.buildable),
+    skipTsConfig: false,
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/generators/library/library.ts:71]: I believe I made this change because the tsconfig.json files were not being created but we then needed them.

I'll check if setting it back to true causes any issues.

**Review by leosvelperez — APPROVED** [review]
LGTM!!! 🎉
